src/Conversion/ONNXToKrnl/Tensor/Resize.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Resize.cpp - Lowering Resize Op
+//-------------------===//
+//
+// Copyright 2019 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX Resize Operator to Krnl dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp"
+
+using namespace mlir;
+
+struct ONNXResizeOpLowering : public ConversionPattern {
+  ONNXResizeOpLowering(MLIRContext *ctx)
+      : ConversionPattern(mlir::ONNXResizeOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    // Gather info.
+    Location loc = op->getLoc();
+    Value alloc;
+    bool insertDealloc = checkInsertDealloc(op);
+    ONNXResizeOp resizeOp = llvm::cast<ONNXResizeOp>(op);
+
+    // Check implementation constraints
+    if (resizeOp.mode() != "nearest" ||
+        resizeOp.coordinate_transformation_mode() != "asymmetric" ||
+        resizeOp.nearest_mode() != "floor")
+      llvm_unreachable("not implemented yet");
```

Comment:
When we don't support something, don't we want to give a more friendly error message? Also, I am not sure what remains from such messages in the Release mode. Please use an emitError message